JHipster 7 beta.0 - mvn verify fails

##### **Overview of the issue**

`mvn verify` fails for newly generated app

```
...
[ERROR] Tests run: 13, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 5.665 s <<< FAILURE! - in com.okta.developer.store.web.rest.ProductResourceIT
...
[ERROR] com.okta.developer.store.web.rest.ProductResourceIT.checkPriceIsRequired  Time elapsed: 0.03 s  <<< FAILURE!
java.lang.AssertionError:
Status expected:<400 BAD_REQUEST> but was:<500 INTERNAL_SERVER_ERROR>
...
[ERROR] Tests run: 9, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.148 s <<< FAILURE! - in com.okta.developer.store.web.rest.errors.ExceptionTranslatorIT
...
[ERROR]   ProductResourceIT.checkPriceIsRequired:173 Status expected:<400 BAD_REQUEST> but was:<500 INTERNAL_SERVER_ERROR>
...
[ERROR]   ProductResourceIT.checkTitleIsRequired:152 Status expected:<400 BAD_REQUEST> but was:<500 INTERNAL_SERVER_ERROR>
...
[ERROR]   ExceptionTranslatorIT.testMethodArgumentNotValid:59 JSON path "$.message" expected:<error.validation> but was:<error.http.500>
...
[ERROR] Tests run: 36, Failures: 3, Errors: 0, Skipped: 0
```

##### **Motivation for or Use Case**

Preparing for a demo.

##### **Reproduce the error**

1. generate app from jdl: https://raw.githubusercontent.com/mraible/choose-jhipster/main/apps.jdl
2. `cd store && ./mvnw verify`

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

##### **JHipster Version(s)**

7.0.0.beta.0

##### **JHipster configuration**

```
application {
  config {
    applicationType gateway
    baseName gateway
    clientFramework vue
    reactive true
    packageName com.okta.developer.gateway
    authenticationType oauth2
    prodDatabaseType postgresql
    serviceDiscoveryType eureka
    testFrameworks [cypress]
  }
  entities Blog, Post, Tag, Product
}

application {
  config {
    applicationType microservice
    baseName blog
    reactive true
    packageName com.okta.developer.blog
    authenticationType oauth2
    prodDatabaseType mysql
    serverPort 8081,
    serviceDiscoveryType eureka
  }
  entities Blog, Post, Tag
}

application {
  config {
    applicationType microservice
    baseName store
    reactive true
    packageName com.okta.developer.store
    authenticationType oauth2
    databaseType mongodb
    devDatabaseType mongodb
    prodDatabaseType mongodb
    enableHibernateCache false
    serverPort 8082
    serviceDiscoveryType eureka
  }
  entities Product
}

entity Blog {
  name String required minlength(3)
  handle String required minlength(2)
}

entity Post {
  title String required
  content TextBlob required
  date Instant required
}

entity Tag {
  name String required minlength(2)
}

entity Product {
  title String required
  price BigDecimal required min(0)
  image ImageBlob
}

relationship ManyToOne {
  Blog{user(login)} to User
  Post{blog(name)} to Blog
}

relationship ManyToMany {
  Post{tag(name)} to Tag{post}
}

paginate Post, Tag with infinite-scroll
paginate Product with pagination

microservice Product with store
microservice Blog, Post, Tag with blog

deployment {
  deploymentType docker-compose
  appsFolders [gateway, blog, store]
  dockerRepositoryName "mraible"
}

```


##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<!--
If the error is during an entity creation or associated with a specific entity.
If you are using JDL, please share that configuration as well.
-->

##### **Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

-   [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

It looks like it works for the `gateway` project (which is reactive + postgresql). 

```
[INFO] Built image to Docker daemon as gateway
[INFO]
[INFO] A new version of Jib (2.7.1) is available (currently using 2.7.0). Update your build configuration to use the latest features and fixes!
[INFO] https://github.com/GoogleContainerTools/jib/blob/master/jib-maven-plugin/CHANGELOG.md
[INFO] Please see https://github.com/GoogleContainerTools/jib/blob/master/docs/privacy.md for info on disabling this update check.
[INFO]
[INFO] Executing tasks:
[INFO] [==============================] 100.0% complete
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
```

I changed the `blog` project to use Couchbase and it fails even worse.
```
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'dependencies.dependency.version' for com.couchbase.client:encryption:jar is missing. @ line 143, column 21
 @
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]
[ERROR]   The project com.okta.developer.blog:blog:0.0.1-SNAPSHOT (/Users/mraible/dev/choose-jhipster/blog/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for com.couchbase.client:encryption:jar is missing. @ line 143, column 21
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
```

I can confirm the `store` project fails with `./mvnw -ntp -Pprod verify jib:dockerBuild`:

```
[ERROR] Tests run: 9, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.166 s <<< FAILURE! - in com.okta.developer.store.web.rest.errors.ExceptionTranslatorIT
[ERROR] com.okta.developer.store.web.rest.errors.ExceptionTranslatorIT.testMethodArgumentNotValid  Time elapsed: 0.027 s  <<< FAILURE!
java.lang.AssertionError: JSON path "$.message" expected:<error.validation> but was:<error.http.500>
	at com.okta.developer.store.web.rest.errors.ExceptionTranslatorIT.testMethodArgumentNotValid(ExceptionTranslatorIT.java:59)

[INFO]
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR]   ProductResourceIT.checkPriceIsRequired:173 Status expected:<400 BAD_REQUEST> but was:<500 INTERNAL_SERVER_ERROR>

> POST /api/products
> WebTestClient-Request-Id: [1]
> Content-Type: [application/json]
> Content-Length: [68]

{"title":"AAAAAAAAAA","image":"AA==","imageContentType":"image/jpg"}

< 500 INTERNAL_SERVER_ERROR Internal Server Error
< Vary: [Origin, Access-Control-Request-Method, Access-Control-Request-Headers]
< Content-Type: [application/problem+json]
< Content-Length: [221]
< Cache-Control: [no-cache, no-store, max-age=0, must-revalidate]
< Pragma: [no-cache]
< Expires: [0]
< X-Content-Type-Options: [nosniff]
< X-XSS-Protection: [1 ; mode=block]
< Feature-Policy: [geolocation 'none'; midi 'none'; sync-xhr 'none'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; fullscreen 'self'; payment 'none']
< Content-Security-Policy: [default-src 'self'; frame-src 'self' data:; script-src 'self' 'unsafe-inline' 'unsafe-eval' https://storage.googleapis.com https://www.google-analytics.com; style-src 'self' 'unsafe-inline'; img-src 'self' data: https://www.google-analytics.com; font-src 'self' data:]
< Referrer-Policy: [strict-origin-when-cross-origin]

221 bytes of content.

[ERROR]   ProductResourceIT.checkTitleIsRequired:152 Status expected:<400 BAD_REQUEST> but was:<500 INTERNAL_SERVER_ERROR>

> POST /api/products
> WebTestClient-Request-Id: [1]
> Content-Type: [application/json]
> Content-Length: [57]

{"price":0,"image":"AA==","imageContentType":"image/jpg"}

< 500 INTERNAL_SERVER_ERROR Internal Server Error
< Vary: [Origin, Access-Control-Request-Method, Access-Control-Request-Headers]
< Content-Type: [application/problem+json]
< Content-Length: [221]
< Cache-Control: [no-cache, no-store, max-age=0, must-revalidate]
< Pragma: [no-cache]
< Expires: [0]
< X-Content-Type-Options: [nosniff]
< X-XSS-Protection: [1 ; mode=block]
< Feature-Policy: [geolocation 'none'; midi 'none'; sync-xhr 'none'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; fullscreen 'self'; payment 'none']
< Content-Security-Policy: [default-src 'self'; frame-src 'self' data:; script-src 'self' 'unsafe-inline' 'unsafe-eval' https://storage.googleapis.com https://www.google-analytics.com; style-src 'self' 'unsafe-inline'; img-src 'self' data: https://www.google-analytics.com; font-src 'self' data:]
< Referrer-Policy: [strict-origin-when-cross-origin]

221 bytes of content.

[ERROR]   ExceptionTranslatorIT.testMethodArgumentNotValid:59 JSON path "$.message" expected:<error.validation> but was:<error.http.500>
[INFO]
[ERROR] Tests run: 36, Failures: 3, Errors: 0, Skipped: 0
```

--- a/generators/entity-server/templates/src/main/java/package/common/field_validators.ejs
+++ b/generators/entity-server/templates/src/main/java/package/common/field_validators.ejs
@@ -28,7 +28,7 @@ if (field.fieldValidate === true) {
 
     if (rules.includes('required') && !isBlob) {
         // reactive tests need a default validation message because lookup is blocking
-        validators.push('@NotNull' + reactive ? '(message = "must not be null")' : '');
+        validators.push('@NotNull' + (reactive ? '(message = "must not be null")' : ''));
     }
     if (rules.includes('minlength') && !rules.includes('maxlength')) {
         validators.push('@Size(min = ' + field.fieldValidateRulesMinlength + ')');

--- a/generators/entity-server/templates/src/main/java/package/common/field_validators.ejs
+++ b/generators/entity-server/templates/src/main/java/package/common/field_validators.ejs
@@ -27,7 +27,8 @@ if (field.fieldValidate === true) {
     const isBlob = field.fieldType === 'byte[]';
 
     if (rules.includes('required') && !isBlob) {
-        validators.push('@NotNull');
+        // reactive tests need a default validation message because lookup is blocking
+        validators.push('@NotNull' + reactive ? '(message = "must not be null")' : '');
     }
     if (rules.includes('minlength') && !rules.includes('maxlength')) {
         validators.push('@Size(min = ' + field.fieldValidateRulesMinlength + ')');

--- a/generators/entity-server/templates/src/main/java/package/domain/Entity.java.ejs
+++ b/generators/entity-server/templates/src/main/java/package/domain/Entity.java.ejs
@@ -221,7 +221,7 @@ public class <%= asEntity(entityClass) %> implements Serializable {
         if (fieldValidateRules.includes('unique')) {
             unique = true;
         } _%>
-    <%- include('../common/field_validators', {field}); -%>
+    <%- include('../common/field_validators', {field, reactive}); -%>
     <%_ } _%>
     <%_ if (!hasDto && typeof field.javadoc != 'undefined') { _%>
     @ApiModelProperty(value = "<%- formatAsApiDescription(field.javadoc) %>"<% if (required) { %>, required = true<% } %>)

--- a/generators/entity-server/templates/src/main/java/package/service/dto/EntityDTO.java.ejs
+++ b/generators/entity-server/templates/src/main/java/package/service/dto/EntityDTO.java.ejs
@@ -78,7 +78,7 @@ _%>
         if (field) {
             if (field.fieldValidate === true) {
     _%>
-    <%- include('../../common/field_validators', {field}); -%>
+    <%- include('../../common/field_validators', {field, reactive}); -%>
     <%_     }
         }
     _%>

--- a/generators/server/templates/src/test/java/package/web/rest/errors/ExceptionTranslatorTestController.java.ejs
+++ b/generators/server/templates/src/test/java/package/web/rest/errors/ExceptionTranslatorTestController.java.ejs
@@ -74,7 +74,7 @@ public class ExceptionTranslatorTestController {
 
     public static class Test<%= dtoSuffix %> {
 
-        @NotNull
+        @NotNull<% if (reactive) { %>(message = "must not be null")<% } %>
         private String test;
 
         public String getTest() {

